src/main/scala/nl.knaw.dans.easy.deposit/Errors.scala

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (C) 2018 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.deposit
+
+import java.nio.file.Path
+import java.util.UUID
+
+import better.files.File
+import nl.knaw.dans.easy.deposit.docs.StateInfo.State.State
+import nl.knaw.dans.easy.deposit.servlets.contentTypePlainText
+import nl.knaw.dans.lib.error._
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.eclipse.jetty.http.HttpStatus._
+import org.scalatra.servlet.FileItem
+import org.scalatra.{ ActionResult, InternalServerError }
+
+import scala.util.Try
+
+object Errors extends DebugEnhancedLogging {
+
+  case class ConfigurationException(msg: String) extends IllegalArgumentException(s"Configuration error: $msg")
+
+  abstract sealed class ServletResponseException(status: Int, httpResponseBody: String)
+    extends Exception(httpResponseBody) {
+    def getActionResult: ActionResult = ActionResult(status, httpResponseBody, Map(contentTypePlainText))
+  }
+
+  implicit class TriedActionResult(val t: Try[ActionResult]) extends AnyVal {
+    def getOrRecoverWithActionResult: ActionResult = t.getOrRecover {
+      case e: ServletResponseException => e.getActionResult
+      case e =>
+        logger.error(s"Not expected exception: ${ e.getMessage }", e)
+        InternalServerError("Internal Server Error", Map(contentTypePlainText))
+    }
+  }
+
+  case class OverwriteException(httpResponseBody: String)
+    extends ServletResponseException(CONFLICT_409, httpResponseBody)
+
+  case class InvalidResourceException(httpResponseBody: String)
+    extends ServletResponseException(NOT_FOUND_404, httpResponseBody)
+
+  case class CorruptDepositException(user: String, id: String, cause: Throwable)
+    extends Exception(s"Invalid deposit uuid $id for user $user: ${ cause.getMessage }", cause)
```

Comment:
shouldn't this be some kind of `ServletResponseException`?

Reply:
From what I understand, you created a superclass `NotExpectedException` for `CorruptDepositException ` now, which does not extend from `ServletResponseException`? Does this mean the `CorruptDepositException` the exception is caught in:

https://github.com/DANS-KNAW/easy-deposit-api/blob/33cfadddd01f9be86dcd4b475ed6a4ef46c47044/src/main/scala/nl.knaw.dans.easy.deposit/Errors.scala#L49-L51

Reply:
Indeed, `ServletResponseException` has no cause argument for more logging than the standard servlet response logging.